source/common/upstream/upstream_impl.cc

```diff
@@ -1298,15 +1298,21 @@ ClusterInfoImpl::ClusterInfoImpl(
   if (http_protocol_options_) {
     Http::FilterChainUtility::FiltersList http_filters = http_protocol_options_->http_filters_;
     has_configured_http_filters_ = !http_filters.empty();
+    static const std::string& upstream_codec_type_url =
```

Comment:
`static const std::string` no the `&`

Reply:
interesting, I borrow this code from https://github.com/envoyproxy/envoy/blob/9dfa144e5e516872eacb79899d4eff1422332f07/source/common/config/utility.h#L302. But `GetTypeName` appears to return `std::string`. Should we fix it everywhere?

Reply:
`static const std::string&` is same with `static const absl::string_view`. Although it is safe in this scenario, but is not a good practice, IMO.

So, I will say, yeah, please fix them.

Reply:
Agree, it's safer since we can't promise protobuf won't change the descriptor symbol storage. Updated.